core/domain/platform_parameter_domain.py

```diff
@@ -35,3 +63,721 @@ class PlatformParameterChange(change_domain.BaseChange):
+                    self._app_version, APP_VERSION_WITH_HASH_REGEXP))
+
+        if self._user_locale not in ALLOWED_USER_LOCALES:
+            raise utils.ValidationError(
+                'Invalid user locale \'%s\', must be one of %s.' % (
+                    self._user_locale, ALLOWED_USER_LOCALES))
+
+        if self._server_mode not in ALLOWED_SERVER_MODES:
+            raise utils.ValidationError(
+                'Invalid server mode \'%s\', must be one of %s.' % (
+                    self._server_mode, ALLOWED_SERVER_MODES))
+
+    @classmethod
+    def from_dict(cls, client_context_dict, server_context_dict):
+        """Creates a new EvaluationContext object by combining both client side
+        and server side context.
+
+        Args:
+            client_context_dict: dict. The client side context.
+            server_context_dict: dict. The server side context.
+
+        Returns:
+            EvaluationContext. The corresponding EvaluationContext domain
+            object.
+        """
+        return cls(
+            client_context_dict.get('client_type'),
+            client_context_dict.get('browser_type'),
+            client_context_dict.get('app_version'),
+            client_context_dict.get('user_locale'),
+            server_context_dict.get('server_mode'),
+        )
+
+
+class PlatformParameterFilter(python_utils.OBJECT):
+    """Domain object for filters in platform parameters."""
+
+    SUPPORTED_FILTER_TYPES = [
+        'server_mode', 'user_locale', 'client_type', 'browser_type',
+        'app_version',
+    ]
+
+    SUPPORTED_OP_FOR_FILTERS = {
+        'server_mode': ['='],
+        'user_locale': ['='],
+        'client_type': ['='],
+        'browser_type': ['='],
+        'app_version': ['=', '<', '<=', '>', '>='],
+    }
+
+    def __init__(self, filter_type, conditions):
+        self._type = filter_type
+        self._conditions = conditions
+
+    @property
+    def type(self):
+        """Returns filter type.
+
+        Returns:
+            str. The filter type.
+        """
+        return self._type
+
+    @property
+    def conditions(self):
+        """Returns filter conditions.
+
+        Returns:
+            list((str, str)). The filter conditions. Each element of the list
+            is a 2-tuple (op, value), where op is the operator for comparison,
+            value is the value used for comparison.
+        """
+        return self._conditions
+
+    def evaluate(self, context):
+        """Tries to match the given context with the filter against its
+        value(s).
+
+        Args:
+            context: EvaluationContext. The context for evaluation.
+
+        Returns:
+            bool. True if the filter is matched.
+        """
+        for op, value in self._conditions:
+            is_matched = self._evaluate_single_value(op, value, context)
+            if is_matched:
+                return True
+        return False
```

Comment:
Slightly more elegant version
```suggestion
        return any(
            self._evaluate_single_value(op, value, context)
            for op, value in self._conditions
        )
```

Reply:
Done.